src/data/nat/choose/multinomial.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2022 Pim Otte. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pim Otte
+-/
+
+import algebra.big_operators.fin
+import algebra.big_operators.order
+import data.nat.choose.basic
+import data.fin.vec_notation
+
+import tactic.linarith
+
+/-!
+# Multinomial
+
+This file defines the multinomial coefficient and several small lemma's for manipulating it.
+
+## Main declarations
+
+- `nat.multinomial`: the multinomial coefficient
+
+-/
+
+open_locale nat
+open_locale big_operators
+
+namespace nat
+
+variables {α : Type*} (s : finset α) (f : α → ℕ)
+variables {a b : α}
+
+/-- The multinomial coefficient. Gives the number of strings consisting of symbols
+from `s`, where `c ∈ s` appears with multiplicity `f c`.
+
+Defined as `(∑ i in s, f i)! / ∏ i in s, (f i)!`
+-/
+def multinomial : ℕ := (∑ i in s, f i)! / ∏ i in s, (f i)!
+
+lemma prod_factorial_dvd_factorial_sum : (∏ i in s, (f i)!) ∣ (∑ i in s, f i)! :=
```

Comment:
This lemma can go in another file I think, probably where factorial is defined.

Reply:
Trying to add this to the file where factorial is defined caused a circular import for the big_operators, so I created a new file for it.